Cache/Hashing.lean

```diff
@@ -11,8 +11,32 @@ namespace Cache.Hashing
 
 open System IO
 
-/-- We cache the hash of each file for faster lookup -/
-abbrev HashM := StateT IO.HashMap IO
+structure HashMemo where
+  depsMap : Lean.HashMap FilePath (Array FilePath)
+  hashMap : HashMap
+  deriving Inhabited
+
+/-- Tells whether a string matches a pattern with at most one wildcard '*' -/
+def matchesAnyPattern (str : String) : List String → IO Bool
+  | [] => pure false
+  | p :: ps => match p.splitOn "*" with
+    | [pat] => if str.startsWith pat then pure true else matchesAnyPattern str ps
+    | [b, e] => if str.startsWith b && str.endsWith e then pure true else matchesAnyPattern str ps
+    | _ => throw $ IO.userError s!"Invalid pattern: {p}"
```

Comment:
The shell does (a better version) of this matching for you on Unix; so this is going to match against the already-matched filenames, which is not good.

Reply:
This is a redo for Unix systems but when it also covers it when it's not done automatically. `*Stuff.lean`, for example, is not caught by Unix